KinKalGeom/inc/SurfaceId.hh

```diff
@@ -0,0 +1,51 @@
+//
+//  Define identifiers for the reconstruction surfaces
+//  original author: David Brown (LBNL) 2023
+//
+#ifndef KinKalGeom_SurfaceId_hh
+#define KinKalGeom_SurfaceId_hh
+#include <map>
+#include <vector>
+#include <string>
+#include "Offline/GeneralUtilities/inc/EnumToStringSparse.hh"
+
+namespace mu2e {
+  class SurfaceIdDetail {
+    public:
+      enum enum_type {
+        unknown =-1,
+        TT_Front=0, TT_Mid, TT_Back, TT_Inner, TT_Outer, // tracker VD equivalents
+        DS_Front=80, DS_Back, DS_Inner, DS_Outer,
+        IPA, OPA, TSDA, // Absorbers in the DS.  These are the inner surfaces for the OPA and TSDA
+        ST_Front=100,ST_Back, ST_Inner, ST_Outer, ST_Foils, // stopping target bounding surfaces and foils
+        TCRV=200 // CRV test planes
+      };
+
+    static std::string const& typeName();
+    static std::map<enum_type,std::string> const& names();
+  };
+  using SurfaceIdEnum = EnumToStringSparse<SurfaceIdDetail>;
+  class SurfaceId {
+    public:
+      // copy the constructors
```

Comment:
Have you considered forwarding the enum_type from SurfaceIdDettail or SurfaceIdEnum ( via using or typedef?)? That way when you need to use the enum names, you can access them as SurfaceId::TT_Mid etc.  

For example at line 33, the type of the first argument would just be SurfaceId::enum_type.